fw/http.c

```diff
@@ -3029,6 +3029,109 @@ tfw_h1_rewrite_purge_to_get(struct sk_buff *head)
 	return -ENOMEM;
 }
 
+#define CHOP_LEADING_CRLF
+#ifdef CHOP_LEADING_CRLF
+/**
+ * Removes single leading CR or CRLF or LF, if any.
+ * Indicates a buggy message if there are more leading CR or LF or other characters <= 0x20 with EBADMSG.
+ * @return zero on success and error code on error.
+ */
+static int
+tfw_h1_chop_leading_crlf(struct sk_buff **head)
+{
+	struct sk_buff *skb = *head;
+	struct sk_buff *stop = *head;
+	char *p, *q;
+	int i, n, ret, ba;
+	char buf[sizeof(__u32)];
+	struct skb_shared_info *si;
+	skb_frag_t *frag;
+
+	if (likely(skb_headlen(skb) >= 3))
+	{
+		p = q = skb->data;
+		if (unlikely(*p == \r)) p++;
+		if (unlikely(*p == \n)) p++;
+		if (unlikely(*p <= 0x20))
+			//Drop erroneous request
+			return EBADMSG;
+		if (unlikely((n = p - q)))
+		{
+			// Cleaning the msg
+			ret = ss_skb_chop_head_tail(skb, skb, n, 0);
+			if (ret)
+				return ret;
+		}
+		return 0;
+	}
+
+	// grab 3 bytes
+	n = 0; *(__u32*)&buf = 0;
```

Comment:
Really?!

Reply:
Yes, understand. Should be `*(__u32*)buf`

Reply:
Yes, because `buf` is already a pointer (pointers and arrays are equal in C), so if you want to zero the whole 4 bytes you should just do `*(__u32*)buf = 0`.

Anyway, please publish for reviewing the code which is compilable and tested. The special case for drafts when we need to discuss the architecture, but in this case it must be stated explicitly that the code is just a pseudocode for discussion.